xformers/triton/k_flash_attn.py

```diff
@@ -0,0 +1,1013 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All rights reserved.
+#
+# This source code is licensed under the BSD license found in the
+# LICENSE file in the root directory of this source tree.
+
+# CREDITS: HazyResearch flash attention.
```

Comment:
Once again, were there any modifications to the original implementation? This would make it easier for us to track what has changed (if anything)

Reply:
Not any major modifications, perhaps easier to just use their implementation directly. Unless we want to do our own autotuning for example. I'll use theirs for now.

Reply:
Actually an issue with this is that `forward_no_grad` no longer works using the  HazyResearch implementation directly, since passing in ctx=None isn't supported. I guess I should revert the latest commit and include the code instead of calling into their API in this case?